service/controller/v11/cluster_resource_set.go

```diff
@@ -63,6 +66,9 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 	if config.CertsSearcher == nil {
 		return nil, microerror.Maskf(invalidConfigError, "config.CertsSearcher must not be empty")
 	}
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.G8sClient must not be empty")
+	}
```

Comment:
Can we use the `%T.G8sClient` foo here? I find it easier to read in errors.

Reply:
Let's do that in a separate PR. This is just a copy/paste from the previous line.